### DISCUSSION TITLE
STM discovery target

### DISCUSSION THREAD

**Discussion by mrZkak** [iNavFlight/inav#8674] [General]
Hello!

I'm  a student and started to build STM32 flight controller. 
As a preparation, I wanted to work with INAV FW on STM discovery board and to start understanding some modules of the code. 
I thought of F7 based MCU but wanted to know that it is supported. Appreciate your help :)
Could you recommend one? 
(I'm working with VScode IDE)

Thanks
Zkak

**Comment by synersignart** [comment]
So far you need to find the Data sheet of the Discovery board online and mark its timers and channels
like most guys they would find the Target board with the closest match in the to the Discovery board in the sec/main.target
you also need to identify its PLL frequency and your choice of SPI and I2C sensors you want to throw in. basicaly laying out the discovery board as a flight controller. its a bit of a learning curve but let me know

**Reply by mrZkak** [reply]
Thanks @synersignart 
I thought to start small by using eval_board and a single peripheral sensor(Gyro MPU6000 for example).
So I thought of two options where I start from FC and try to find a board as you said:

1) Diatone Mamba F722 - iNav target: /src/main/MAMBAF722_2022A -> eval_board: Nucleo F722ZE
2) Diatone Mamba F405 MK3.5 - iNav target: /src/main/MAMBA F405_2022A -> eval_board: WaveShare Core405R

For both I've verified that the MCU version is exact and I2C/SPI pins are open for peripherals. 
What do I need to check more? (I thought of memory map but didn't find info)
Thanks
Zkak

Links:
https://www.st.com/en/evaluation-tools/nucleo-f722ze.html
https://stm32-base.org/boards/STM32F405RGT6-WaveShare-CorexxxR

**Reply by synersignart** [reply]
apparently the PLL Parameters relating to the Frequency of this board  by dafult INAV use 24mhz but some dev boards runs off 25mhz

**Reply by mrZkak** [reply]
Which board did you mean? 
Please let me know where did you find it and I will look at the register to check whether the PLL is configurable...

**Reply by synersignart** [reply]
oh first you need to know the data spects of your discovery board you plan to use and its frequency

**Reply by synersignart** [reply]
like the channels ,Timers ,UARTs and other Data pin maps this way it can be evaluated

**Reply by mrZkak** [reply]
Thank you, I will look into the specs and update
Zkak

**Comment by mrZkak** [comment]
@DzikuVx 
Hi Pawel, in parallel - do you work with a discovery / eval board and can give a recommendation?
Thanks
Zkak
